src/macro/macroApi.ml

```diff
@@ -1477,7 +1477,7 @@ let decode_type_def v =
 	) in
 	(* if our package ends with an uppercase letter, then it's the module name *)
 	let pack,name = (match List.rev pack with
-		| last :: l when not (is_lower_ident last) -> List.rev l, last
+		| last :: l when StringHelper.starts_uppercase last -> List.rev l, last
```

Comment:
A question here is if "not lower" is equal to "upper". If it's not we're slightly changing semantics here, which may or may not be intended.

Reply:
It does change them, but only for invalid packages like `"0"` and `""` I think? The comment above even says "if our package ends with an uppercase letter", so the previous check seems a bit awkward.